### PR TITLE
Some optimisations

### PR DESCRIPTION
**Pull request by wixzi**
# Summary

Some optimisations

<table>
  <tr>
    <td>
    Before
    </td>
    <td>
    After
    </td>
  </tr> 
  <tr>
    <td>
    Screenshot 2022-08-31 at 20 01 09
    </td>
    <td>
    Screenshot 2022-08-31 at 20 02 59
    </td>
  </tr>
</table>

# To Test

Everything should work as expected

# Background

### CODE REVIEW

File: src/components/swap/SwapButtons/SwapButton.tsx [hunk 1]
```diff
@@ -14,6 +13,14 @@ import {
...
 import { ButtonPrimary } from '../../Button'
 
+function shuffle<T>(array: T[]) {
+  for (let i = array.length - 1; i > 0; i--) {
+    let j = Math.floor(Math.random() * (i + 1)) // random index from 0 to i
+    ;[array[i], array[j]] = [array[j], array[i]]
+  }
+  return array
+}
+
```
Comment (berteotti) [inline src/components/swap/SwapButtons/SwapButton.tsx:23]: Should we add this to a separate file under `/utils`? It looks odd in here

**Review by berteotti — COMMENTED** [review]
Good cleanup, don't forget to remove `lodash` from the dependencies.

File: src/constants/index.tsx [hunk 2]
```diff
@@ -21,7 +21,7 @@ import { ReactNode } from 'react'
 import RightArrow from '../assets/images/arrow-right.svg'
 import BaoswapLogo from '../assets/images/baoswap-logo.png'
 import Coinbase from '../assets/images/coinbase.svg'
-import CurveLogo from '../assets/images/curve-logo.svg'
+import CurveLogo from '../assets/images/curve-logo.png'
 import DFYNLogo from '../assets/images/dfyn-logo.svg'
 import LevinswapLogo from '../assets/images/levinswap-logo.svg'
 import ZeroXLogo from '../assets/images/logos/ZeroX.svg'
```
Comment (adamazad) [inline src/constants/index.tsx:27]: Couldn't we compress the curve logo svg?

Comment (wixzi) [reply src/constants/index.tsx:27]: I tried that first. But the curve logo is a huge SVG with so many gradient effects. A compressed version itself is very big and we use only a 32*32 size of it in swapr buttons and marquee.

Comment (wixzi) [reply src/constants/index.tsx]: Added a compressed version of curve-logo as well.

File: src/components/Routing/BaseRedirect.tsx [hunk 3]
```diff
@@ -1,12 +1,12 @@
 import { Navigate, useParams, useSearchParams } from 'react-router-dom'
 
 // Redirects from the /swap/:outputCurrency path to the /swap?outputCurrency=:outputCurrency format
-export function RedirectToSwap() {
+export function BaseRedirect({ pathname = '/swap' }) {
   const [search] = useSearchParams()
   const { outputCurrency } = useParams<{ outputCurrency: string }>()
```
Comment (adamazad) [inline src/components/Routing/BaseRedirect.tsx:6]: Lovely!

**Review by adamazad — APPROVED** [review]

**Comment by MilanVojnovic95** [comment]
Tested and looks ok to me 👍

**Review by berteotti — APPROVED** [review]
Great job!

**Pull request merged**